### ISSUE TITLE
PARAMETER_DEF in FinalLocalVariable doesn't seem to work in interfaces

### ISSUE THREAD

**Issue by Machine-Maker** [checkstyle/checkstyle#16081]
https://checkstyle.org/checks/coding/finallocalvariable.html#FinalLocalVariable

```bash
/var/tmp $ javac test.java
# success

/var/tmp $ cat checkstyle.xml
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="FinalLocalVariable">
      <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
    </module>
  </module>
</module>

/var/tmp $ cat YOUR_FILE.java
interface Test {
  static void test(String param) {
    String local = "";
    System.out.println(param);
  }
}

/var/tmp $ java -jar checkstyle-10.21.0-all.jar -c checkstyle.xml test.java
Starting audit...
[ERROR] /Users/jake/Projects/Minecraft/PaperMC/test.java:3:12: Variable 'local' should be declared final. [FinalLocalVariable]
Audit done.
Checkstyle ends with 1 errors.
```

---

The `param` local variable in the static method inside the Test interface should be reported as needing the `final` modifier which, if I'm understanding the documentation correctly, adding the `PARAMETER_DEF` token to the `FinalLocalVariable` check should do.

If I change `Test` to a `class` instead of `interface`, it correctly reports both, so I'm guessing someone just forgot about methods inside of interfaces.

**Comment by romani** [comment]
Yes, code in interface is recently new feature of java :)

**Comment by mahfouz72** [comment]
Just to make it clear abstract methods in the interface still should be ignored, and only the `default` and `static` methods (methods that can have a body) should be checked. Methods with bodies (static, default or private) were introduced in Java 8 and 9.

```Java
interface test {
    void test(int a);   // ok, this is an abstract method we should not violate parameter 'a'

    default void test2(int b) { // violation, b should be final
            // body
    }

    static void test3(int c) { // violation, c should be final
         // body
    }
}
```

**Comment by romani** [comment]
Fix is merged
